### DISCUSSION TITLE
Howto mount additional sshfs for clients during boot?

### DISCUSSION THREAD

**Discussion by AllesMeins** [ltsp/ltsp#623] [Q&A] [answered]
On my server there are some additional filesystems I want to make available to the clients (/media/VolumeGroups). I've already overcome some obstacles but haven't been able to get it to work. Since I'm suspecting that this isn't such an unusual usecase: Does anybody has some best practices how to realize an additional sshfs mount during boot?

My current config:
```
[common]
OMIT_IMAGE_EXCLUDES="media/*"
ADD_IMAGE_EXCLUDES="media/VolumeGroups/*"
[clients]
FSTAB_1="sshfs#USER@server:/media/VolumeGroups /media/VolumeGroups  fuse IdentityFile=/home/USER/.ssh/id_rsa,UserKnownHostsFile=/home/USER/.ssh/known_hosts,uid=1000,gid=1000,users,idmap=user,noatime,allow_other,_netdev,reconnect,ro 0 0"
```

The first part is my crude attempt to provide the client with a empty /media/VolumGroups folder (there probably is a better way to do this?) which seems to work. The second part is my fstab-line which works when I test it on the client with a manual "#sudo mount -a" after boot, but fails during boot-up (without any meaningful error in journalct I could find). I guess maybe the network or the /home-directories aren't available yet? So long story short: Since this seems to get more and more "hackier" - what is the proper way to do it?

I'm running a chrootless ltsp on Ubuntu 20.04.

**Comment by alkisg** [comment, marked as answer]
The first part could by done by POST_INIT_MKDIR_MEDIA="mkdir -p /media/VolumeGroups".

The second part isn't appropriate for multi-user systems. User mounts should go under /media/user, not under a global directory. Don't use SSHFS for this, use e.g. NFSv4 or wireguard.

For automatic per-user SSHFS mounts, you can either use PAM, or you can create an /etc/xdg/autostart script, that does something like this:

```shell
# This just sets up ssh keys for passwordless ssh
ltsp remoteapps true
mkdir ~/mnt
sshfs user@server:/path ~/mnt
```

**Reply by AllesMeins** [reply]
Thank you for your quick answer! I was able to adjust it to my needs and now everything works as hoped!
